src/main/io/osd.c

```diff
@@ -770,10 +770,47 @@ static bool osdDrawSingleElement(uint8_t item)
             STATIC_ASSERT(OSD_FORMAT_MESSAGE_BUFFER_SIZE <= sizeof(buff), osd_warnings_size_exceeds_buffer_size);
 
             const batteryState_e batteryState = getBatteryState();
+            const timeUs_t currentTimeUs = micros();
```

Comment:
What is the reason for moving this up?

Reply:
Fair enough. The reason I am bringing this up is because the ordering dictates what takes priority when elements overlap.

Reply:
This is all one element- the warning element.  But it has priorities as well and that’s the primary reason for the changes.  Previously the arming disabled was almost at the end so things like “LOW BATTERY” would display instead even though arming failed.

The cycling through multiple arming disabled reasons is just a bonus.

Reply:
Ah yes, you're right. Thinking of it, the readability of this could probably be improved by refactoring it into multiple functions.